### PR TITLE
Build: add git pre-commit hooks

### PR DESCRIPTION
**Pull request by omgaz**
- Protect `master` branch from being directly committed to unless a release.
 - Run `dist` to build source files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,6 +37,8 @@ Development
 
 -	Install NPM dependencies: `npm i`
 
+-   Set-up git hooks `rm -rf .git/hooks && cd .git/ && ln -s ../scripts/git-hooks hooks && chmod +x hooks/* && cd -` (optional)
```
Comment (camjc) [inline README.md]: spacing inconsistent. remove `(optional)`

File: scripts/git-hooks/pre-commit [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+PATH="node_modules/.bin:$PATH";
```
Comment (jharrington22) [inline scripts/git-hooks/pre-commit]: Don't need the trailing ; js boys.

**Comment by bchew** [comment]
LGTM

**Comment by camjc** [comment]
LGTM. Merging!

**Pull request merged**